#
# This patch enables the usage of external CFLAGS
# (supplied via 'make CFLAGS="-O99'')
# without destrying the internal needed ones, by renaming the internals
#
diff -ruN ppp-2.4.1-orig/chat/Makefile.linux ppp-2.4.1/chat/Makefile.linux
--- ppp-2.4.1-orig/chat/Makefile.linux	Fri Aug 13 03:54:32 1999
+++ ppp-2.4.1/chat/Makefile.linux	Sat Sep 27 02:32:04 2003
@@ -6,17 +6,20 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-O2 -g -pipe $(CDEFS)
+C_FLAGS=	-O2 -g -pipe $(CDEFS) $(CFLAGS)
 
 INSTALL= install
 
 all:	chat
 
+%.o: %.c
+	$(CC) $(C_FLAGS) -c -o $@ $<
+
 chat:	chat.o
 	$(CC) -o chat chat.o
 
 chat.o:	chat.c
-	$(CC) -c $(CFLAGS) -o chat.o chat.c
+	$(CC) -c $(C_FLAGS) -o chat.o chat.c
 
 install: chat
 	mkdir -p $(BINDIR)
diff -ruN ppp-2.4.1-orig/pppd/Makefile.linux ppp-2.4.1/pppd/Makefile.linux
--- ppp-2.4.1-orig/pppd/Makefile.linux	Thu Mar  8 06:00:35 2001
+++ ppp-2.4.1/pppd/Makefile.linux	Sat Sep 27 02:35:21 2003
@@ -58,62 +58,65 @@
 
 COMPILE_FLAGS= -D_linux_=1 -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MULTILINK -DHAVE_MMAP
 
-CFLAGS= $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS)
+C_FLAGS= $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS) $(CFLAGS)
 
 ifdef CHAPMS
-CFLAGS   += -DCHAPMS=1
+C_FLAGS   += -DCHAPMS=1
 ifndef USE_CRYPT
 LIBS     := -ldes $(LIBS)
 else
-CFLAGS   += -DUSE_CRYPT=1
+C_FLAGS   += -DUSE_CRYPT=1
 ifneq ($(wildcard /usr/include/crypt.h),)
-CFLAGS   += -DHAVE_CRYPT_H=1
+C_FLAGS   += -DHAVE_CRYPT_H=1
 endif
 endif
 PPPDOBJS += md4.o chap_ms.o
 ifdef MSLANMAN
-CFLAGS   += -DMSLANMAN=1
+C_FLAGS   += -DMSLANMAN=1
 endif
 endif
 
 ifdef HAS_SHADOW
-CFLAGS   += -DHAS_SHADOW
+C_FLAGS   += -DHAS_SHADOW
 #LIBS     := -lshadow $(LIBS)
 endif
 
 # For "Pluggable Authentication Modules", see ftp.redhat.com:/pub/pam/.
 ifdef USE_PAM
-CFLAGS   += -DUSE_PAM
+C_FLAGS   += -DUSE_PAM
 LIBS     := -lpam -ldl $(LIBS)
 endif
 
 # Lock library binary for Linux is included in 'linux' subdirectory.
 ifdef LOCKLIB
 LIBS     := -llock $(LIBS)
-CFLAGS   += -DLOCKLIB=1
+C_FLAGS   += -DLOCKLIB=1
 endif
 
 ifdef PLUGIN
-CFLAGS	+= -DPLUGIN
+C_FLAGS	+= -DPLUGIN
 LDFLAGS	+= -Wl,-E
 LIBS	+= -ldl
 endif
 
 ifdef FILTER
 LIBS    += -lpcap
-CFLAGS  += -DPPP_FILTER -I/usr/include/pcap
+C_FLAGS  += -DPPP_FILTER -I/usr/include/pcap
 endif
 
 ifdef HAVE_INET6
      PPPDSRCS += ipv6cp.c eui64.c
      HEADERS  += ipv6cp.h eui64.h
      PPPDOBJS += ipv6cp.o eui64.o
-     CFLAGS   += -DINET6=1
+     C_FLAGS   += -DINET6=1
 endif
 
 
 INSTALL= install -o root
 
+%.o: %.c
+	$(CC) $(C_FLAGS) -c -o $@ $<
+
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
@@ -122,10 +125,10 @@
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
+	$(CC) $(C_FLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
 
 clean:
 	rm -f $(PPPDOBJS) pppd *~ #* core
 
 depend:
-	$(CPP) -M $(CFLAGS) $(PPPDSRCS) >.depend
+	$(CPP) -M $(C_FLAGS) $(PPPDSRCS) >.depend
diff -ruN ppp-2.4.1-orig/pppd/plugins/Makefile.linux ppp-2.4.1/pppd/plugins/Makefile.linux
--- ppp-2.4.1-orig/pppd/plugins/Makefile.linux	Wed Mar  7 05:21:18 2001
+++ ppp-2.4.1/pppd/plugins/Makefile.linux	Sat Sep 27 02:34:37 2003
@@ -1,15 +1,15 @@
 CC	= gcc
-CFLAGS	= -g -O2 -I.. -I../../include -fPIC
+C_FLAGS	= -g -O2 -I.. -I../../include -fPIC $(CFLAGS)
 LDFLAGS	= -shared
 INSTALL	= install
 
 all:	minconn.so passprompt.so
 
 minconn.so: minconn.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) minconn.c
+	$(CC) -o $@ $(LDFLAGS) $(C_FLAGS) minconn.c
 
 passprompt.so: passprompt.c
-	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) passprompt.c
+	$(CC) -o $@ $(LDFLAGS) $(C_FLAGS) passprompt.c
 
 LIBDIR	= /usr/lib/pppd
 
diff -ruN ppp-2.4.1-orig/pppdump/Makefile.linux ppp-2.4.1/pppdump/Makefile.linux
--- ppp-2.4.1-orig/pppdump/Makefile.linux	Mon Jul 26 13:09:29 1999
+++ ppp-2.4.1/pppdump/Makefile.linux	Sat Sep 27 02:25:27 2003
@@ -1,10 +1,13 @@
-CFLAGS= -O -I../include/net
+C_FLAGS= -O -I../include/net $(CFLAGS)
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
 
 all:	pppdump
 
+%.o: %.c
+	$(CC) $(C_FLAGS) -c -o $@ $<
+
 pppdump: $(OBJS)
 	$(CC) -o pppdump $(OBJS)
 
@@ -14,4 +17,4 @@
 install:
 	mkdir -p $(BINDIR) $(MANDIR)/man8
 	$(INSTALL) -s -c pppdump $(BINDIR)
-	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)/man8
+	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)/man8
\ No newline at end of file
diff -ruN ppp-2.4.1-orig/pppstats/Makefile.linux ppp-2.4.1/pppstats/Makefile.linux
--- ppp-2.4.1-orig/pppstats/Makefile.linux	Wed Mar 25 03:21:19 1998
+++ ppp-2.4.1/pppstats/Makefile.linux	Sat Sep 27 02:26:23 2003
@@ -13,7 +13,7 @@
 
 INSTALL= install -o root -g daemon
 
-CFLAGS = $(COPTS) $(COMPILE_FLAGS)
+C_FLAGS = $(COPTS) $(COMPILE_FLAGS) $(CFLAGS)
 
 all: pppstats
 
@@ -22,11 +22,11 @@
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)/man8/pppstats.8
 
 pppstats: $(PPPSTATSRCS)
-	$(CC) $(CFLAGS) -o pppstats pppstats.c $(LIBS)
+	$(CC) $(C_FLAGS) -o pppstats pppstats.c $(LIBS)
 
 clean:
 	rm -f pppstats *~ #* core
 
 depend:
-	cpp -M $(CFLAGS) $(PPPSTATSRCS) >.depend
-#	makedepend $(CFLAGS) $(PPPSTATSRCS)
+	cpp -M $(C_FLAGS) $(PPPSTATSRCS) >.depend
+#	makedepend $(C_FLAGS) $(PPPSTATSRCS)
